store/sqlstore/file_info_store.go

```diff
@@ -95,6 +95,72 @@ func (fs SqlFileInfoStore) Get(id string) (*model.FileInfo, *model.AppError) {
 	return info, nil
 }
 
+func (fs SqlFileInfoStore) GetWithOptions(page, perPage uint, opt *model.GetFileInfosOptions) ([]*model.FileInfo, *model.AppError) {
+	if perPage == 0 || page == 0 {
```

Comment:
I think `page == 0` should be a valid input, since we use 0 based semantics everywhere else.

Reply:
I noticed that we use `int` everywhere else for `page` and `perPage`. I've used `uint` here as a "free" compile time check. Postgres and MySql don't allow negative offset or limit. I think a long time ago Postgres would silently take negative offsets as zero.

Should I change to `int` to be consistent with other parts of the code?

Reply:
Yes, that's an excellent call -- let's switch to `int`.